ballerina/http_connection.bal

```diff
@@ -184,6 +186,24 @@ public isolated client class Caller {
     }
 }
 
+isolated function retrieveMediaTypeForApplicationError(string? annotatedMediaType, map<string|string[]> headers) returns string? {
+    if annotatedMediaType is string {
+        return annotatedMediaType;
+    } else {
+        if !headers.hasKey(CONTENT_TYPE) {
+            return;
+        } 
+        string[]|string mediaType = headers.get(CONTENT_TYPE);
+        if mediaType is string {
+            return mediaType;
+        } else {
```

Comment:
There can't be an array of media types right ?

Reply:
Fixed with [9a0b4f39dab43eda54e01b64619dd0baaa658eaf](https://github.com/ballerina-platform/module-ballerina-http/pull/839/commits/9a0b4f39dab43eda54e01b64619dd0baaa658eaf)

Reply:
When setting `content-type` header to error-details it is set as a `string[]`. Hence reverted the fix.